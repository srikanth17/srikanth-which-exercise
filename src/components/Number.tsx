import React from 'react';

interface NumberProps {
  isActive: boolean;
  number: number;
  handleSelected: (num: number) => void;
}

const Number = ({ isActive, number, handleSelected }: NumberProps) => (
  <label
    className={`number${isActive ? ' active' : ''}`}
    data-testid="number-label"
  >
    <input
      checked={isActive}
      name="number"
      type="checkbox"
      value={number}
      onChange={(evt: any) => {
        handleSelected(number);
      }}
    />
    {number}
  </label>
);

export default Number;
