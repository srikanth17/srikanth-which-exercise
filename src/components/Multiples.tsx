import React, { useState } from 'react';
import Number from './Number';
import { isMultipleOf } from '../utils/utils';

interface MultiplesProps {
  numbers: number[];
}

const Multiples = ({ numbers }: MultiplesProps) => {
  const [selectedNumber, setSelectedNumber] = useState<number | undefined>(
    undefined
  );

  function handleSelected(num: number) {
    if (selectedNumber) {
      setSelectedNumber(num);
      return;
    }

    setSelectedNumber(num); // for the first selection only
  }

  return (
    <div className="grid">
      {numbers.map(number => (
        <div key={number} data-testid="number">
          <Number
            isActive={Boolean(
              selectedNumber && isMultipleOf(selectedNumber, number)
            )}
            number={number}
            handleSelected={handleSelected}
          />
        </div>
      ))}
    </div>
  );
};

export default Multiples;
