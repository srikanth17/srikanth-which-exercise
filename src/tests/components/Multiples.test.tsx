import React from 'react';
import { shallow } from 'enzyme';
import Multiples from '../../components/Multiples';
import Number from '../../components/Number';

const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

test('should render Multiples correctly', () => {
  const wrapper = shallow(<Multiples numbers={numbers} />);
  expect(wrapper).toMatchSnapshot();
});

test('renders all Number', () => {
  const wrapper = shallow(<Multiples numbers={numbers} />);
  expect(wrapper.find(Number)).toHaveLength(numbers.length);
});
