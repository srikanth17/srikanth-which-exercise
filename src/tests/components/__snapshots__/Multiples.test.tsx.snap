// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Multiples correctly 1`] = `
<div
  className="grid"
>
  <div
    data-testid="number"
    key="1"
  >
    <Number
      handleSelected={[Function]}
      isActive={false}
      number={1}
    />
  </div>
  <div
    data-testid="number"
    key="2"
  >
    <Number
      handleSelected={[Function]}
      isActive={false}
      number={2}
    />
  </div>
  <div
    data-testid="number"
    key="3"
  >
    <Number
      handleSelected={[Function]}
      isActive={false}
      number={3}
    />
  </div>
  <div
    data-testid="number"
    key="4"
  >
    <Number
      handleSelected={[Function]}
      isActive={false}
      number={4}
    />
  </div>
  <div
    data-testid="number"
    key="5"
  >
    <Number
      handleSelected={[Function]}
      isActive={false}
      number={5}
    />
  </div>
  <div
    data-testid="number"
    key="6"
  >
    <Number
      handleSelected={[Function]}
      isActive={false}
      number={6}
    />
  </div>
  <div
    data-testid="number"
    key="7"
  >
    <Number
      handleSelected={[Function]}
      isActive={false}
      number={7}
    />
  </div>
  <div
    data-testid="number"
    key="8"
  >
    <Number
      handleSelected={[Function]}
      isActive={false}
      number={8}
    />
  </div>
  <div
    data-testid="number"
    key="9"
  >
    <Number
      handleSelected={[Function]}
      isActive={false}
      number={9}
    />
  </div>
  <div
    data-testid="number"
    key="10"
  >
    <Number
      handleSelected={[Function]}
      isActive={false}
      number={10}
    />
  </div>
</div>
`;
